Environment_Engine/Query/PointsMatch.cs

```diff
@@ -51,7 +51,8 @@ public static bool PointsMatch(this List<Point> controlPoints, List<Point> measu
             foreach (Point p in controlPoints)
             {
                 Point ptInMeasure = measurePoints.Where(x => x.X == p.X && x.Y == p.Y && x.Z == p.Z).FirstOrDefault();
```

Comment:
not related tot he PR, but spotted scrolling through. Could be safer doing this with a distance and tolerance to avoid equal for doubles? Maybe with a square distance even.
```
                Point ptInMeasure = measurePoints.Where(x => p.SquareDistance(x) < squareTolerance).FirstOrDefault();
```

where the tolerance is input/defined above to whatever is suitable. (Please ignore if to off-topic for this PR)

Reply:
Thanks for the spot @IsakNaslundBh - have modified accordingly! :smile: